gguf-py/scripts/gguf-dump.py

```diff
@@ -101,25 +101,200 @@ def dump_metadata_json(reader: GGUFReader, args: argparse.Namespace) -> None:
+            elif curr_type in reader.gguf_scalar_to_np:
+                value = str(field.parts[-1][0])
+        else:
+            if field.types[0] == GGUFValueType.ARRAY:
+                curr_type = field.types[1]
+                if curr_type == GGUFValueType.STRING:
+                    render_element = min(5, total_elements)
+                    for element_pos in range(render_element):
+                        value += repr(str(bytes(field.parts[-1 - element_pos]), encoding='utf-8')[:5]) + (", " if total_elements > 1 else "")
+                elif curr_type in reader.gguf_scalar_to_np:
+                    render_element = min(7, total_elements)
+                    for element_pos in range(render_element):
+                        value += str(field.parts[-1 - element_pos][0]) + (", " if total_elements > 1 else "")
+                value = f'[ {value}{" ..." if total_elements > 1 else ""} ]'
+        markdown_content += f'| {n:3} | {pretty_type:10} | {total_elements:8} | {field.name:38} | {value:<78} |\n'
+
+    markdown_content += "\n"
+
+    if not args.no_tensors:
+        # Group tensors by their prefix and maintain order
+        tensor_prefix_order = []
+        tensor_name_to_key = {}
+        tensor_groups = {}
+        total_elements = sum(tensor.n_elements for tensor in reader.tensors)
+
+        for key, tensor in enumerate(reader.tensors):
+            tensor_components = tensor.name.split('.')
+            tensor_prefix = tensor_components[0]
+
+            if tensor_prefix == 'blk':
+                tensor_prefix = f"{tensor_components[0]}.{tensor_components[1]}"
+
+            if tensor_prefix not in tensor_groups:
+                tensor_groups[tensor_prefix] = []
+                tensor_prefix_order.append(tensor_prefix)
+
+            tensor_name_to_key[tensor.name] = key
+            tensor_groups[tensor_prefix].append(tensor)
+
+        # Tensors Mapping Dump
+        markdown_content += f'## Tensors Overview {element_count_rounded_notation(total_elements)} Elements\n'
+        markdown_content += f'Total number of elements in all tensors: {total_elements} Elements\n'
+        markdown_content += '\n'
+
+        for group in tensor_prefix_order:
+            tensors = tensor_groups[group]
+            group_elements = sum(tensor.n_elements for tensor in tensors)
+            markdown_content += f"- [{translate_tensor_name(group)} Tensor Group - {element_count_rounded_notation(group_elements)} Elements](#{group.replace('.', '_')})\n"
+
+        markdown_content += "\n"
+
+        for group in tensor_prefix_order:
+            tensors = tensor_groups[group]
+            group_elements = sum(tensor.n_elements for tensor in tensors)
+            group_percentage = group_elements / total_elements * 100
+            markdown_content += f"### {translate_tensor_name(group)} Tensor Group : {element_count_rounded_notation(group_elements)} Elements <a name=\"{group.replace('.', '_')}\"></a>\n"
```

Comment:
The anchor tag could probably be replaced by `{#name_here}`, which is the syntax for special attributes (see <https://michelf.ca/projects/php-markdown/extra/#spe-attr>), and it's supported by a lot of Markdown parsers, even `lowdown`.

```suggestion
            markdown_content += f"### {translate_tensor_name(group)} Tensor Group : {element_count_rounded_notation(group_elements)} Elements {{#{group.replace('.', '_')}}}\n"
```

An empty anchor feels weird.

Reply:
Investigated and saw that it's not widely supported and also more importantly not supported in github. I've adjusted the anchor to cover the title at least.